libexec/bats-core/bats-format-junit

```diff
@@ -74,8 +74,13 @@ print_test_case() {
     if [[ -n "$_system_out_log" ]]; then
       printf "        <system-out>%s</system-out>\n" "$(xml_escape "${_system_out_log}")"
     fi
-    if [[ -n "$_buffer_log" || "$test_result_state" == not_ok ]]; then
+    if [[ "$test_result_state" == not_ok ]]; then # Failed tests need a <failure> element.
+      # If we have system err output, we use that as the failure text.
       printf "        <failure type=\"failure\">%s</failure>\n" "$(xml_escape "${_buffer_log}")"
+    else
+      if [[ -n "$_buffer_log" ]]; then
+        printf "        <system-err>%s</system-err>\n" "$(xml_escape "${_buffer_log}")"
+      fi
```

Comment:
This is the core of my thinking here: only "not_ok" tests should report a `<failure`... element.
Everything else should say it's ok while recording as much of the test's output as system-out / system-err as possible.